src/Cohesity.Powershell/Scripts/Restore/Restore-CohesityFileV2.ps1

```diff
@@ -0,0 +1,405 @@
+            }
+            if($restoreMethod -ne 'ExistingAgent')  {
+
+                $restoreParams.vmwareParams.recoverFileAndFolderParams.vmwareTargetParams.originalTargetConfig["targetVmCredentials"] = $vmCredentials
+            }
+        }
+
+        # find files for restore
+        foreach($file in $files)  {
+
+            if($noIndex)  {
+
+                if($file[-1] -eq '/')  {
+
+                    $isDirectory = $True
+                }
+                else  {
+
+                    $isDirectory = $false
+                }
+                $restoreParams.vmwareParams.recoverFileAndFolderParams.filesAndFolders += @{
+                    "absolutePath" = $file;
+                    "isDirectory" = $isDirectory
+                }
+            }
+            else  {
+
+                $searchParams = @{
+                    "fileParams" = @{
+                        "searchString"       = $file;
+                        "sourceEnvironments" = @(
+                            "kVMware"
+                        );
+                        "objectIds"          = @(
+                            $objectId
+                        )
+                    };
+                    "objectType" = "Files"
+                }
+                $url = "/v2/data-protect/search/indexed-objects"
+                $searchJson = $searchParams | ConvertTo-Json -Depth 100
+                $search = Invoke-RestApi -Method Post -Uri $url -Body $searchJson
+
+                $thisFile = $search.files | Where-Object {("{0}/{1}" -f $_.path, $_.name) -eq $file -or ("{0}/{1}/" -f $_.path, $_.name) -eq $file}
+                if(!$thisFile)  {
+
+                    $errorMsg = "file $file not found"
+                    Write-Output $errorMsg
+                    CSLog -Message $errorMsg
+
+                }
+                else  {
+                    if($file[-1] -eq '/')  {
+
+                        $isDirectory = $True
+                        $absolutePath = "{0}/{1}/" -f $thisFile[0].path, $thisFile[0].name
+                    }
+                    else  {
+
+                        $isDirectory = $false
+                        $absolutePath = "{0}/{1}" -f $thisFile[0].path, $thisFile[0].name
+                    }
+                    $restoreParams.vmwareParams.recoverFileAndFolderParams.filesAndFolders += @{
+                        "absolutePath" = $absolutePath;
+                        "isDirectory" = $isDirectory
+                    }
+                }
+            }
+        }
+
+        # perform restore
+        if($restoreParams.vmwareParams.recoverFileAndFolderParams.filesAndFolders.Count -gt 0)  {
+
+            $url = "/v2/data-protect/recoveries"
+            $restoreJson = $restoreParams | ConvertTo-Json -Depth 100
+            $restoreTask = Invoke-RestApi -Method Post -Uri $url -Body $restoreJson
+            $restoreTaskId = $restoreTask.id
+            if($wait)  {
+
+                Start-Sleep 3
```

Comment:
Not sure why sleeps are needed ? The poll function below takes care of it anyways.

Reply:
Add it as static variable : $PollingForAPI = 5,

Reply:
I noticied this. After getting the response in line 377, job status "Running" is being found after some delay. Hence this is required. Otherwise "Running" state will be not detected at all by the script due to backend limitation

Reply:
You are right, we have seen this in other integrations as well. Probably add it as a comment ?

Reply:
Constant is introduced